docs/devguide/flashfirmware.rst

```diff
@@ -20,39 +36,72 @@ Start the build with either yotta::
 
   yt build
 
-...or use the Makefile::
+...or use the makefile::
 
   make all
 
-The result is a microbit-micropython hex file (i.e. ``microbit-micropython.hex``)
-found in the build/bbc-microbit-classic-gcc-nosd/source from the root of the
-repository.
+The result is a microbit-micropython hex file
+(i.e. ``microbit-micropython.hex``) found in the
+build/bbc-microbit-classic-gcc-nosd/source from the root of the repository.
 
-The Makefile does some extra preprocessing of the source, which is needed only
-if you add new interned strings to ``qstrdefsport.h``. The Makefile also puts
+The Makefile does some extra preprocessing of the source, which is needed
+only if you add new interned strings to qstrdefsport.h. The Makefile also puts
 the resulting firmware at build/firmware.hex, and includes some convenience
 targets.
 
 Preparing firmware and a Python program
----------------------------------------
+=======================================
+
+Using ``tools/makecombinedhex.py`` you can combine the MicroPython firmware
+with a Python script and produce a hex file ready for uploading to the
+micro:bit.::
+
+  ./makecombinedhex.py <firmware.hex> <script.py> [-o <combined.hex>]
+  # Output goes to stdout if no filename is given.
+
+Using ``tools/hexlify.py`` you can turn a Python script into Intel HEX format
+to be concatenated at the end of the MicroPython firmware.hex.  A simple header
+is added to the script.::
+
+  ./hexlifyscript.py <script.py>
+  
+It also accepts data on standard input.
+
+
+micro:bit V2
+============
+
+This applies to MicroPython for the micro:bit V2, the source of which can be 
+found here: `microbit-foundation/micropython-microbit-v2 <https://github.com/microbit-foundation/micropython-microbit-v2>`_.
+
+The repository also contains a history of 
+`MicroPython firmware builds <https://github.com/microbit-foundation/micropython-microbit-v2/actions>`_.
+
+Dependencies
+------------
+
+- `cMake <https://cmake.org/>`_
+- `git <https://git-scm.com/>`_ 
+- `python <https://www.python.org/downloads/>`_
 
-tools/makecombined
+Clone the repository::
 
-hexlify
+    $ git clone https://github.com/microbit-foundation/micropython-microbit-v2
 
+Update the submodules::
 
+    $ git submodule update --init
```

Comment:
I would combine these steps to be able to add the "cd micropython-microbit-v2" as well.


```suggestion
    $ git clone https://github.com/microbit-foundation/micropython-microbit-v2
    $ cd micropython-microbit-v2
    $ git submodule update --init
```